app/code/community/Retargeting/Tracker/controllers/ProductsController.php

```diff
@@ -10,35 +10,32 @@
 class Retargeting_Tracker_ProductsController extends Mage_Core_Controller_Front_Action
 {
     public function indexAction() {
-		echo '<?xml version="1.0" encoding="UTF-8"?>';
-		echo '
-		<products>';
-
-		$params = $this->getRequest()->getParams();
-
-        if ( isset($params['key']) && $params['key'] != '' && $params['key'] == Mage::getStoreConfig('retargetingtracker_options/token/token') ) {
-
-			$collection = Mage::getResourceModel('catalog/product_collection');
-			foreach($collection as $product) {
-				$product = Mage::getModel('catalog/product')->load($product->getId());
-				$product_price = Mage::helper('tax')->getPrice($product, $product->getPrice());
-				$product_promo = ( Mage::helper('tax')->getPrice($product, $product->getPrice()) - Mage::helper('tax')->getPrice($product, $product->getFinalPrice()) > 0 ? Mage::helper('tax')->getPrice($product, $product->getFinalPrice()) : 0 );
-				$product_image = ( $product->getThumbnail() != 'no_selection' ? htmlspecialchars(Mage::helper('catalog/image')->init($product, 'thumbnail')) : htmlspecialchars(Mage::helper('catalog/image')->init($product, 'image')->resize(500)) );
-				$product_url = $product->getProductUrl();
-				echo '
-			<product>
-				<id>'.$product->getId().'</id>
-				<stock>'.$product->getIsInStock().'</stock>
-				<price>'.$product_price.'</price>
-				<promo>'.$product_promo.'</promo>
-				<url>'.$product_url.'</url>
-				<image>'.$product_image.'</image>
-			</product>';
-			}
-		
-		}
-
-		echo '
-		</products>';
+    
+    			$productCollection = Mage::getModel('catalog/product')->getCollection();
+          $productCollection->addAttributeToSelect(array('name', 'price', 'special_price'));
+          $retargetingFeed = array();
+          
+    			foreach($productCollection as $product) {
+            
+            $retargetingFeed[] = array(
+                'id' => $product->getId(),
+                'price' => (double)$product->getPrice(),
+                'promo' =>( $product->getPrice() - $product->getFinalPrice() > 0 ) ?  (double)$product->getFinalPrice() : 0,
+                'promo_price_end_date' => $product->getSpecialToDate(),
```

Comment:
Aici trebuie testat mai amanuntit pentru ca desi ii setezi produsului "Special Price To Date" in admin, setarea trebuie sa vizibila si pe front. Asta nu am reusit sa fac ca nu am gasit acea setare.